internal/metamorph/processor_helpers.go

```diff
@@ -1,7 +1,78 @@
 package metamorph
 
-func (p *Processor) GetProcessorMapSize() int {
+import (
+	"github.com/bitcoin-sv/arc/internal/metamorph/store"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+)
 
+func (p *Processor) GetProcessorMapSize() int {
 	return p.ProcessorResponseMap.Len()
+}
+
+func updateStatusMap(statusUpdatesMap map[chainhash.Hash]store.UpdateStatus, statusUpdate store.UpdateStatus) {
+	foundStatusUpdate, found := statusUpdatesMap[statusUpdate.Hash]
+
+	if !found || (found && shouldUpdateStatus(statusUpdate, foundStatusUpdate)) {
+		if len(statusUpdate.CompetingTxs) > 0 {
```

Comment:
Wouldn't this be a bit more explicit?

status.Status == metamorph_api.Status_DOUBLE_SPEND_ATTEMPTED

Reply:
Here I followed the same reasoning as I explained above. Let me know what you think